diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd0b60a..6343ca5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,7 @@ add_executable(SDLJumpStart
 	src/engine.cpp
 	src/font.cpp
 	src/main.cpp
+	src/sdl_utils.cpp
 )
 
 # Do you need C++11? Maybe just indicate the features you need like this
diff --git a/src/AssertMsg.cpp b/src/AssertMsg.cpp
index 61d056f..17a6007 100644
--- a/src/AssertMsg.cpp
+++ b/src/AssertMsg.cpp
@@ -1,4 +1,5 @@
 #include <iostream>
+#include "sdl_gl.h"
 
 void CustomAssertMsg(const char* expr_str, bool expr, const char* file, int line, const char* msg)
 {
@@ -7,6 +8,6 @@ void CustomAssertMsg(const char* expr_str, bool expr, const char* file, int line
         std::cerr << "Assert failed:\t" << msg << "\n"
             << "Expected:\t" << expr_str << "\n"
             << "Source:\t\t" << file << ", line " << line << "\n";
-        abort();
+        sdl_application_abort("Application quit ...");
     }
 }
diff --git a/src/engine.cpp b/src/engine.cpp
old mode 100755
new mode 100644
index f33c43a..aa495b4
--- a/src/engine.cpp
+++ b/src/engine.cpp
@@ -77,7 +77,7 @@ Engine::Engine()
 	glLoadIdentity( );
 	gluPerspective( 60.0, (float)clientW/(float)clientH, 0.00000000001, 1024.0 );
 	
-	font = new FixedWidthBMPFont("images/font.bmp", 16);
+	font = new FixedWidthBMPFont("../images/font.bmp", 16);
 
 
 	// TODO: More initialization calls ehre.
@@ -108,7 +108,6 @@ void Engine::Update()
 	// Too lazy to do this properly, but want consistent simulation results because I was testing the stability of the integrator.
 	elapsedtime = 0.016;
 
-
 	// TODO: Make update calls here.
 
 
diff --git a/src/engine.h b/src/engine.h
old mode 100755
new mode 100644
diff --git a/src/font.cpp b/src/font.cpp
old mode 100755
new mode 100644
index 8ffb166..8c89234
--- a/src/font.cpp
+++ b/src/font.cpp
@@ -12,7 +12,7 @@
 #include <GL/glu.h>
 #endif  // Anything other than __APPLE__
 
-#include <string>
+#include <string.h>
 #include <sstream>
 
 //  The * aread makes up GLYPH_PERCENT_OF_CELL_X (% of the square)
diff --git a/src/font.h b/src/font.h
old mode 100755
new mode 100644
diff --git a/src/main.cpp b/src/main.cpp
old mode 100755
new mode 100644
index 55b6b6d..5833d1b
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -1,5 +1,5 @@
 #include "engine.h"
-
+#include "sdl_gl.h"
 #include "SDL.h"
 
 int main(int argc, char *argv[])
@@ -8,7 +8,7 @@ int main(int argc, char *argv[])
 	try
 	{
 		engine = new Engine();
-
+		getInfo();
 		for(;;)
 		{
 			engine->Update();
diff --git a/src/sdl_gl.h b/src/sdl_gl.h
index d32eafa..2d03b02 100644
--- a/src/sdl_gl.h
+++ b/src/sdl_gl.h
@@ -10,9 +10,13 @@
 #ifdef _WIN32
 #include <windows.h>
 #endif  // _WIN32
+
 #include <GL/gl.h>
 #include <GL/glu.h>
 #endif  // Anything other than __APPLE__
 
+void sdl_application_abort( const char *);
+void getInfo(void);
+void getTextureInfo( SDL_Texture *);
 
-#endif  // SDL_OPEN_GL_H
\ No newline at end of file
+#endif  /* SDL_OPENGL_H */
\ No newline at end of file
diff --git a/src/sdl_gl_texture.cpp b/src/sdl_gl_texture.cpp
old mode 100755
new mode 100644
index 46a491d..09c87a2
--- a/src/sdl_gl_texture.cpp
+++ b/src/sdl_gl_texture.cpp
@@ -20,7 +20,7 @@ unsigned int LoadTexture(const char *filename, bool mipmapped)
 	SDL_Surface *surface;
 	surface = SDL_LoadBMP(filename);
 	if (surface == NULL) {
-		fprintf(stderr, "Failed to load texture: %s", filename);
+		fprintf(stderr, "Failed to load texture: %s \n", filename);
 		return 0;
 	}
 	
diff --git a/src/sdl_gl_texture.h b/src/sdl_gl_texture.h
old mode 100755
new mode 100644
